apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/RowAsyncEnumerator.cs

```diff
@@ -0,0 +1,298 @@
+                owner._currentCell.Labels = null;
+
+                // auto transition
+                return NewCell.Instance.HandleChunk(owner, chunk);
+            }
+        }
+
+        private sealed class NewCell : RowMergeState
+        {
+            public static readonly NewCell Instance = new NewCell();
+
+            private NewCell() { }
+
+            public override RowMergeState HandleChunk(RowAsyncEnumerator owner, CellChunk chunk)
+            {
+                // update name & make sure it changed
+                bool idChanged = false;
+                if (chunk.FamilyName != null)
+                {
+                    if (chunk.FamilyName != owner._currentCell.Family?.Name)
+                    {
+                        idChanged = true;
+                        owner._currentCell.Family = new Family { Name = chunk.FamilyName };
+                        owner._currentCell.Row.Families.Add(owner._currentCell.Family);
+                    }
+                    owner.Assert(chunk.Qualifier != null, "NewCell has a familyName, but no qualifier");
+                }
+
+                if (chunk.Qualifier != null && chunk.Qualifier != owner._currentCell.Column?.Qualifier)
+                {
+                    idChanged = true;
+                    owner._currentCell.Column = new Column { Qualifier = chunk.Qualifier };
+                    owner.Assert(
+                        owner._currentCell.Family != null,
+                        "NewCell has a qualifier, but no familyName");
+                    owner._currentCell.Family.Columns.Add(owner._currentCell.Column);
+                }
+
+                idChanged = idChanged ||
+                    owner._currentCell.Timestamp != chunk.TimestampMicros;
+                owner._currentCell.Timestamp = chunk.TimestampMicros;
+
+                idChanged = idChanged ||
+                    !chunk.Labels.SequenceEqual(owner._currentCell.Labels ?? Enumerable.Empty<string>());
+                owner._currentCell.Labels = chunk.Labels;
+
+                owner.Assert(idChanged, "NewCell must have a new identifier");
+                owner.Assert(chunk.Value != null, "NewCell must have a value");
+
+                // calculate cell size
+                owner._currentCell.ValueSizeExpected = chunk.ValueSize > 0 ? chunk.ValueSize : chunk.Value.Length;
+                owner._currentCell.ValueSizeRemaining = owner._currentCell.ValueSizeExpected - chunk.Value.Length;
+
+                // Start building cell
+                owner._currentCell.Value = chunk.Value;
+
+                if (owner._currentCell.ValueSizeRemaining != 0)
+                {
+                    return CellInProgress.Instance;
+                }
+
+                owner.AddCompletedCellToRow();
+                return NewCell.Instance;
+            }
+            
+            public override RowMergeState CommitRow(RowAsyncEnumerator owner)
+            {
+                return NewRow.Instance;
```

Comment:
Validate that we don't have data still to come (via ValueSizeRemaining)?

Reply:
Added a debug assert since it's not possible in this state. We can only be here if a NewCell state processed a complete value or a CellInProgress state processed the completion of an incomplete value.